app/services/broadcasts/welcome_notification/generator.rb

```diff
@@ -11,13 +11,16 @@ def self.call(*args)
       end
 
       def call
-        # This method should find the user based on the `receiver_id`.
-        # It should then determine the appropriate Broadcast for a user,
-        # based on the `created_at` and the different conditions for sending a notification.
-        # `welcome_broadcast = ...`
+        return if commented_on_welcome_thread?
 
-        # Once it has the appropriate Broadcast to be sent, it should send a notification for it:
-        # `Notification.send_welcome_notification(receiver_id, welcome_broadcast.id)`
+        welcome_broadcast = Broadcast.find_by(title: "Welcome Notification")
+        Notification.send_welcome_notification(receiver_id, welcome_broadcast.id)
+      end
+
+      def commented_on_welcome_thread?
+        welcome_threads = Article.where("slug LIKE 'welcome-thread%'")
+        user = User.find_by(id: receiver_id)
+        Comment.where(commentable: welcome_threads, user_id: receiver_id).any? && Time.zone.now >= user.created_at + 3.hours
```

Comment:
Can you tell me more about why we are doing a check for the time/the user’s created at here?

Reply:
Something else to think about: how will this generator handle the other checks for the other welcome notifications? I think this would be a good scenario to abstract out the logic for each notification into separate methods rather than putting it directly into the call method

Reply:
I added the check for the User's creation time because I remembered that we are only sending this Notification to Users who have not commented on a Welcome Thread, three hours after their account creation. I wasn't sure if it should be handled here or in the rake task to come but thought that here would be a good place to start until I got feedback otherwise.